app/experimenter/experiments/forms.py

```diff
@@ -434,19 +431,48 @@ def is_valid(self):
         return super().is_valid() and self.variants_formset.is_valid()
 
     def save(self, *args, **kwargs):
-        experiment = super().save(*args, **kwargs)
+        with transaction.atomic():
+            self.variants_formset.save()
+            return super().save(*args, **kwargs)
```

Comment:
I know it was my suggestion initially, to use a context manager, but another option is to use it as a decorator which is pretty neat because they meat of the method doesn't have to be indented just because of this. And there's really no reason to NOT run the whole method in a single transaction.

```python
    @transaction.atomic
    def save(self, *args, **kwargs):
        self.variants_formset.save()
        return super().save(*args, **kwargs)
```

Reply:
Yeah I saw that too, very convenient!